pkg/reconciler/v1alpha1/route/reconcile_resources.go

```diff
@@ -127,3 +129,32 @@ func (c *Reconciler) updateStatus(ctx context.Context, route *v1alpha1.Route) (*
 	c.Recorder.Eventf(route, corev1.EventTypeNormal, "Updated", "Updated status for route %q", route.Name)
 	return updated, nil
 }
+
+// Update the lastPinned annotation on revisions we target so they don't get GC'd.
+func (c *Reconciler) reconcileTargetRevisions(ctx context.Context, t *traffic.TrafficConfig, route *v1alpha1.Route) error {
+	rev_errs := []error{}
+	for _, target := range t.Targets {
+		for _, rt := range target {
+			tt := rt.TrafficTarget
+			rev, err := c.revisionLister.Revisions(route.Namespace).Get(tt.RevisionName)
+			if apierrs.IsNotFound(err) {
+				c.Logger.Infof("Unable to update lastPinned for missing revision %q", tt.RevisionName)
+				continue
+			} else if err != nil {
+				rev_errs = append(rev_errs, err)
+				continue
+			}
+
+			rev.Annotations["revision.knative.dev/lastPinned"] = fmt.Sprintf("%d", c.clock.Now().Unix())
```

Comment:
This should check for nil `rev.Annotations`.

Reply:
Perhaps we can make this a method on `Revision`?

Reply:
done